http-clients/src/main/java/com/palantir/remoting/http/errors/SerializableErrorErrorDecoder.java

```diff
@@ -112,7 +125,15 @@ private static Exception constructException(Class<? extends Exception> exception
     }
 
     private static Exception constructException(
-            Class<? extends Exception> exceptionClass, String message, Throwable wrappedException) {
+            Class<? extends Exception> exceptionClass, String message, Throwable wrappedException, int status) {
+        if (exceptionClass.equals(ClientErrorException.class)) {
+            return new ClientErrorException(status, wrappedException);
+        } else if (exceptionClass.equals(ServerErrorException.class)) {
+            return new ServerErrorException(status, wrappedException);
+        } else if (exceptionClass.equals(WebApplicationException.class)) {
+            return new WebApplicationException(wrappedException, status);
+        }
```

Comment:
Just tried that: apparently all the WebApplicationExceptions call into a single constructor that takes cause, meaning that they're all setting cause to ``null`` if you don't set it. This prevents us from setting it with ``initCause``, because of the "set once" limitation.